### PR TITLE
dev: parse v0.11.0 DA calldata

### PR DESCRIPTION
**Pull request by drspacemn**
- add support for [v0.11.0](https://docs.starknet.io/documentation/architecture_and_concepts/Network_Architecture/on-chain-data/#v0.11.0_format) manual DA parsing from storage notification stream
- include encoding for DA word encoding
- storage change parsing is done in a `ThinStateDiff` structure (potential to add scale encoding to this api type and save this to the kv db)
- add da config examples
- add cli flag for `--da-conf` as the dev file overrides don't include the da config

```
cargo run --release -- --dev --da-layer ethereum --da-conf examples/da-confs/ethereum.json
```

# Pull Request type

- Bugfix
- Feature [x]
- Code style update (formatting, renaming)
- Refactoring (no functional changes, no API changes)
- Build-related changes
- Documentation content changes
- Testing
- Other (please describe):

## What is the current behavior?

- the DA client currently encodes the prev0.11.0 da format

Resolves: #NA

## What is the new behavior?

- encode the v0.11.0 fmt

## Does this introduce a breaking change?

minor - if use or import of exposed util functions

### CODE REVIEW

File: crates/node/src/commands/run.rs [hunk 1]
```diff
@@ -45,6 +45,10 @@ pub struct ExtendedRunCmd {
     #[clap(long)]
     pub da_layer: Option<DaLayer>,
 
+    /// Choose a supported DA Layer
+    #[clap(long)]
+    pub da_conf: Option<String>,
```
Comment (tdelabro) [inline crates/node/src/commands/run.rs:50]: Make them an arg group. 
The whole group is optional, but as soon as one is used, all must be

File: crates/client/data-availability/src/lib.rs [hunk 2]
```diff
@@ -76,57 +82,82 @@ where
 {
     pub async fn prove_current_block(da_mode: DaMode, client: Arc<C>, madara_backend: Arc<mc_db::Backend<B>>) {
         let mut storage_event_st = client
-            .storage_changes_notification_stream(None, None)
+            .storage_changes_notification_stream(None, Some(&[(SubStorageKey(twox_128(PALLET_STARKNET).into()), None)]))
```
Comment (tdelabro) [inline crates/client/data-availability/src/lib.rs:85]: I think you can go `(None, None)` unti we get some light on https://github.com/paritytech/polkadot-sdk/issues/1989

File: crates/client/data-availability/src/lib.rs [hunk 3]
```diff
@@ -76,57 +82,82 @@ where
...
+
+                if child_key == twox_128(STARKNET_NONCE) {
+                    // collect nonce information in state diff
+                    state_diff
+                        .nonces
+                        .insert(ContractAddress(bytes_to_key(&rest_key)), Nonce(bytes_to_felt(&storage_val)));
+                    accessed_addrs.insert(ContractAddress(bytes_to_key(&rest_key)));
+                } else if child_key == twox_128(STARKNET_STORAGE) {
+                    // collect storage update information in state diff
+                    if rest_key.len() > 32 {
```
Comment (tdelabro) [inline crates/client/data-availability/src/lib.rs:121]: I don't think this could be false.

**Review by tdelabro — CHANGES_REQUESTED** [review]

**Comment by drspacemn** [comment]
holding for work here: https://github.com/keep-starknet-strange/madara/pull/1217

**Pull request closed without merging**